### PR TITLE
Octree performance patches

### PR DESCRIPTION
**Pull request by moneromooo**

### CODE REVIEW

**Comment by eugeneko** [comment]
I don't see any immediate issues, but this change feels risky.
Let's wait for others' feedback.

File: Source/Urho3D/Graphics/Octree.cpp [hunk 1]
```diff
@@ -247,7 +247,7 @@ void Octant::GetDrawablesInternal(OctreeQuery& query, bool inside) const
         query.TestDrawables(start, end, inside);
     }
 
-    for (auto child : children_)
+    for (const auto &child : children_)
```
Comment (1vanK) [inline Source/Urho3D/Graphics/Octree.cpp:250]: Use [Coding conventions](https://urho3d.github.io/documentation/HEAD/_coding_conventions.html)
> Pointers and references append the * or & symbol to the type without a space in between. For example Drawable* drawable, Serializer& dest

Comment (eugeneko) [reply Source/Urho3D/Graphics/Octree.cpp:250]: > children_ is array of pointers. There is no optimization

SharedPtr copying is relatively expensive.
I would go either for `const auto&` or explicit `Octant*`

Comment (1vanK) [reply Source/Urho3D/Graphics/Octree.cpp:250]: Octant* children_[NUM_OCTANTS]{};

**Comment by 1vanK** [comment]
The commits are shipped in bulk at such a speed that I do not believe they are tested in any way)

**Comment by moneromooo** [comment]
Weird. I must have confused it with another. I'll remove that patch then.

File: Source/Urho3D/Graphics/Octree.cpp (hunk 1 above)
Comment (eugeneko) [reply Source/Urho3D/Graphics/Octree.cpp]: Oh, these are plaint pointers. Nasty.
Yep, in this case there's no optimization.

**Comment by moneromooo** [comment]
Tested just for my own needs really. If there's any you don't like, it's not like you have to take them, they're here if you want them only.

**Pull request closed without merging**